src/validation.py

```diff
@@ -0,0 +1,66 @@
+import re
+
+from src.message import Message as message
+from src.user_interface import UserInterface as ui
```

Comment:
I find it odd that this has dependencies on both the Message class and the UserInterface.  These two classes are [plugins to your system](https://github.com/t-keazirian/tic-tac-toe/blob/d6b5c1135f10700bcd472bc3fa6f2b1e9709384f/src/game.py#L9).

In fact, should this validation class depend on anything at all?

Reply:
Hmmm...interesting.  I call the `ui` and the `message` to be passed in in both the `handle_invalid_input` methods - I am assuming there must be another way then to handle the message being displayed without depending on these other classes...?  But that way is not clear to me...

Reply:
I would expect a validation class to simply validate things.

Here is how this might be called form a Game class (its pseudo code to express the idea)
```Python
...
self.ui.display_message(self.message.prompt_for_move(current_player))
next_move = self.ui.get_next_move(self.message.get_move_for_player(self.current_player))
while !self.validator.valid_move(next_move):
    self.ui.display_message(self.message.invalid_move(self))
    next_move = self.ui.get_next_move(self.message.get_move_for_player(self.current_player))
self.board.mark_board(next_move, self.game_board, self.current_player)
 ...
```

Reply:
a validator simply answer the question, "is this valid?" (`True` or `False`)